### ISSUE TITLE
Bug: Published date is one month earlier than actual when it's the first day of the month

### ISSUE THREAD

**Issue by powersagitar** [transitive-bullshit/nextjs-notion-starter-kit#604]
#### Description

As title suggests, the published date is one month earlier than specified.

On Notion:
<img width="400" alt="Screenshot 2024-03-17 at 11 46 38 PM" src="https://github.com/transitive-bullshit/nextjs-notion-starter-kit/assets/114868133/4ab5f537-13fd-4aa5-9e58-2704509a36f1">

On site:
<img width="238" alt="Screenshot 2024-03-17 at 11 47 40 PM" src="https://github.com/transitive-bullshit/nextjs-notion-starter-kit/assets/114868133/3db5b0bf-b441-4ba6-a460-acc37061f76d">

#### Notion Test Page ID
https://powerium.notion.site/Decoding-Conditionals-A-Dive-into-if-else-switch-case-Lookup-Tables-and-Interfaces-19254846bcb64aac9c340e81518a9dac

#### Note
Apparently this issue only happens when published date is the first day of the month. E.g., when the published date is Feb. 1, it appears as Jan. 1; however, when it's Feb. 2, it appears normally as Feb. 2.

Revision: `0e40674e88f8429fbb136d032ae77be17f1bda9c`

**Comment by araza95** [comment]
Certainly, here's your revised message including a request on how to reproduce the bug:

"@powersagitar, Hi, I hope you are doing well. I would like to fix this bug for you.

Can you elaborate on the issue, specifically regarding how the published date is processed and displayed? Also, could you please provide steps on **how to reproduce** this bug? 

Understanding the underlying cause and reproduction steps will help me propose an effective solution. Thank you!"

**Comment by powersagitar** [comment]
Hi @araza95, thank you for your interest.

#### Steps to Reproduce
1. Duplicate the [Notion template](https://transitive-bs.notion.site/Next-js-Notion-Starter-Kit-Template-7875426197cf461698809def95960ebf)
2. Change the `Published` field of one of the posts to `February 1, 2024`
3. Publish Notion page to web
4. Checkout `main` branch of this repo
5. Adjust `rootNotionPageId` in [site.config.ts](https://github.com/transitive-bullshit/nextjs-notion-starter-kit/blob/0e40674e88f8429fbb136d032ae77be17f1bda9c/site.config.ts) accordingly
6. Run `yarn` to install dependencies, `yarn dev` to start a development server

#### The Bug
- After going through above steps, you should be able to observe an unhandled runtime error, suggesting that text content does not match server-rendered URL
  - The error is gone after changing the `Published` field to any date that is not the first day of a month
- As per my findings the cause is that the publish date on the bottom left corner of the blog posts on website homepage does not match the one you entered on Notion
  - Remember you set the `Published` field to `February 1, 2024` in step 2, but on the homepage it appears to be `Jan 1, 2024`, which is one month earlier.
- The bug is present regardless of the month as long as the day is the first day of the month
- If `Published` is set to `January 1, 2024`, the publish date on the website homepage becomes `Dec 1, 2024`

Please let me know if more clarification is required. :)
